### ISSUE TITLE
How to change 

### ISSUE THREAD

**Issue by cmbowyer13** [tensorflow/models#4248]
Please go to Stack Overflow for help and support:

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [3,3,1,32] rhs shape= [3,3,3,32]
         [[Node: save/Assign_28 = Assign[T=DT_FLOAT, _class=["loc:@FeatureExtractor/MobilenetV1/Conv2d_0/weights"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](FeatureExtractor/MobilenetV1/Conv2d_0/weights, save/RestoreV2_28)]]

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).

I want to run inference on grey scale images, and i don't know how to solve this problem to change the depth dimension of the weights for this 4D tensor.  I have successfully done decent inference on rgb images before but now i want to do inferencing on images of the form [none, none, 1] or more so raw data in grey scale format. 

2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**:
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by cmbowyer13** [comment]
Thanks. I have another approach I tried where I just removed the second and third convolutional layers from [3,3,**3**,32] so that it resulted in [3,3,1,32] and resaved the checkpoint file, to have the matrix multiplication work out correctly. However it then complained about the decoding of my raw data which I don't understand. I was able to create TF records when passing in my encoded raw data directly from matrices, but I am not sure what should go in the other attributes of the tf_example object or how to silence anything related to images/ 

def create_tf_example(group,path,files):
    
    with tf.gfile.GFile(os.path.join(path, '{}'.format(group.filename+'.jpg')), 'rb') as fid:
        encoded_jpg = fid.read()
    encoded_jpg = fid.read()
    encoded_jpg_io = io.BytesIO(encoded_jpg)
    image = Image.open(encoded_jpg_io)
    w_jpeg, h_jpeg = image.size
    
    #define raw data height and width
    h_raw = 512
    w_raw = 7000
    
    filename = group.filename.encode('utf8')
    image_format = b'.jpg'
    xmins = []
    xmaxs = []
    ymins = []
    ymaxs = []
    classes_text = []
    classes = []
    
    #Conversion factor from jpeg coordinates to raw data coordinates
    jpeg_to_raw_w = w_raw / w_jpeg
    jpeg_to_raw_h = h_raw / h_jpeg
    
    for index, row in group.object.iterrows():
        xmins.append((row['xmin'] * jpeg_to_raw_w) / w_raw)
        xmaxs.append((row['xmax'] * jpeg_to_raw_w) / w_raw)
        ymins.append((row['ymin'] * jpeg_to_raw_h) / h_raw)
        ymaxs.append((row['ymax'] * jpeg_to_raw_h) / h_raw)
        classes_text.append(str(row['class']).encode('utf8')) 
        classes.append(class_text_to_int(str(row['class'])))

    # Read in data as multi-dimensional arrays
    #print(group.filename)
    
    idx = files.index('/media/ssd/JupyterHubNotebooks/calebbowyer/public_material/models/research/object_detection/Object-Detection/images/train3_raw/' + group.filename + '.npy')
    plot_data = np.load(files[idx])
    plot_data = plot_data.reshape(512,7000,1)
    
    # Encode your matrix input as string
    encoded_inputs = plot_data.tostring()
    print("encoded_inputs is: ", encoded_inputs)
        
    tf_example = tf.train.Example(features=tf.train.Features(feature={
        'image/height': dataset_util.int64_feature(h_raw),
        'image/width': dataset_util.int64_feature(w_raw),
        'image/filename': dataset_util.bytes_feature(filename),
        'image/source_id': dataset_util.bytes_feature(filename),
        'image/encoded': dataset_util.bytes_feature(encoded_inputs), #Stores the raw data!
        'image/format': dataset_util.bytes_feature(image_format),
        'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
        'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
        'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
        'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
        'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
        'image/object/class/label': dataset_util.int64_list_feature(classes),
        'image/channels': dataset_util.int64_feature(1) 
    }))
    

    
    
    return tf_example
But I would like to know a correct way of creating my TF records from the raw data and being able to use object detection on them using the api. Thank you.

**Comment by cmbowyer13** [comment]
Can I get a generic response time please? I would really like to fix this or have this capability in the api.

**Comment by k-w-w** [comment]
There isn't a set response time for questions. You could try posting this question to the general community on [Stack Overflow](http://stackoverflow.com/questions/tagged/tensorflow).

I can try my best to help, although as stated before I'm not familiar with the code. Can you paste the error you mentioned? Also, it would be helpful if you filled out the form posted by tensorflowbutler above:

> What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by serjl** [comment]
You should do two changes of channels (3-->1) in 
\models\research\object_detection\data_decoders\tf_example_decoder.py

(line 189) image_key='image/encoded', format_key='image/format', channels=1)

(line 315) tensor_dict[fields.InputDataFields.image].set_shape([None, None, 1])

**Comment by serjl** [comment]
https://github.com/minhnhat93/tf_object_detection_multi_channels/blob/master/README.md
